src/writer/lsp_manager.py

```
@@ -0,0 +1,171 @@
+                "pylsp command not found. Python LSP features will not be available. "
+                "Install with: pip install python-lsp-server[all]"
+            )
+            return False
+
+        try:
+            self.port = self._find_available_port()
+            logger.info(f"Starting Python LSP server on port {self.port}")
+
+            # Start pylsp with WebSocket support
+            # Note: pylsp --ws may exit when client disconnects, this is expected behavior
+            self.process = subprocess.Popen(
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)